tools/diff-dotgraph.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+import pydot
+import sys
+from typing import Dict, List
+
+
+def usage():
+    usage_text = """Usage: {0} "$(cat graph1)" "$(cat graph2)"
+
+This tool takes two graphs and finds all the nodes in the second graph that
+don't exist in the first graph. All these nodes are then colored in grey and
+the final graph is printed back.
+
+Example: {0} \\
+    "$(git show HEAD^:pkg/defkinds/webserver/testdata/build/from-git-context.json | zbuild llbgraph)" \\
+    "$(cat pkg/defkinds/webserver/testdata/build/from-git-context.json | zbuild llbgraph)" \\
+    | dot /dev/stdin -o /dev/stdout -T png | feh - """
+    
+    print(usage_text.format(sys.argv[0]), file=sys.stderr)
+
+
+def list_edges(graph: pydot.Graph) -> Dict[str, str]:
```

Comment:
what does an edge represent ?

Reply:
Edges are the lines between the nodes. It's the mathematical term in graph theory :wink: